### PR TITLE
T17 events

### LINKED ISSUE #17: Refactor Phalcon\Events
Refactor Phalcon\Events

### PR DESCRIPTION
**Pull request by niden**
Hello!

* Type: new feature
* Link to issue: #17 

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/phalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [ ] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Events

Thanks

### CODE REVIEW

**Comment by niden** [comment]
We can ignore Codacy for now, I will have a run at it after other components are merged

File: src/Events/Event.php [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    }
+
+    /**
+     * Check whether the event is cancelable.
+     *
+     * ```php
+     * if ($event->isCancelable()) {
+     *     $event->stop();
+     * }
+     * ```
```
Comment (zsilbi) [inline src/Events/Event.php:125]: @return is missing

File: src/Events/Event.php [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+     */
+    public function isCancelable(): bool
+    {
+        return $this->cancelable;
+    }
+
+    /**
+     * Check whether the event is currently stopped.
+     */
```
Comment (zsilbi) [inline src/Events/Event.php:134]: `@return` is missing

File: src/Events/Event.php [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        return $this->type;
+    }
+
+    /**
+     * Check whether the event is cancelable.
+     *
+     * ```php
+     * if ($event->isCancelable()) {
+     *     $event->stop();
+     * }
```
Comment (zsilbi) [inline src/Events/Event.php:124]: `@return` is missing

File: src/Events/Event.php (hunk 2 above)
Comment (zsilbi) [inline src/Events/Event.php:138]: `@return` is missing

**Review by zsilbi — APPROVED** [review]

**Pull request merged**